redash/tasks/general.py

```diff
@@ -63,6 +65,28 @@ def send_mail(to, subject, html, text):
         logger.exception("Failed sending message: %s", message.subject)
 
 
+@job("queries", timeout=30, ttl=90)
+def test_connection(data_source_id):
+    try:
+        data_source = models.DataSource.get_by_id(data_source_id)
+        data_source.query_runner.test_connection()
+    except Exception as e:
+        return e
+    else:
+        return True
+
+
+@job("schemas", queue_class=Queue, at_front=True, timeout=30, ttl=90)
```

Comment:
Unfortunately for some data sources timeout needs to be much longer. Let's start with 5 minutes.